src/structures/APIMessage.js

```diff
@@ -0,0 +1,321 @@
+  }
+
+  /**
+   * Whether or not the target is a user.
+   * @type {boolean}
+   * @readonly
+   */
+  get isUser() {
+    const User = require('./User');
+    const GuildMember = require('./GuildMember');
+    return this.target instanceof User || this.target instanceof GuildMember;
+  }
+
+  /**
+   * Makes the content of this message.
+   * @returns {string|string[]}
+   */
+  makeContent() { // eslint-disable-line complexity
+    const GuildMember = require('./GuildMember');
+
+    // eslint-disable-next-line eqeqeq
+    let content = Util.resolveString(this.options.content == null ? '' : this.options.content);
+    const isSplit = typeof this.options.split !== 'undefined' && this.options.split !== false;
+    const isCode = typeof this.options.code !== 'undefined' && this.options.code !== false;
+    const splitOptions = isSplit ? Object.assign({}, this.options.split) : undefined;
+
+    let mentionPart = '';
+    if (this.options.reply && !this.isUser && this.target.type !== 'dm') {
+      const id = this.target.client.users.resolveID(this.options.reply);
+      mentionPart = `<@${this.options.reply instanceof GuildMember && this.options.reply.nickname ? '!' : ''}${id}>, `;
+      if (isSplit) {
+        splitOptions.prepend = `${mentionPart}${splitOptions.prepend || ''}`;
+      }
+    }
+
+    if (content || mentionPart) {
+      if (isCode) {
+        const codeName = typeof this.options.code === 'string' ? this.options.code : '';
+        content = `${mentionPart}\`\`\`${codeName}\n${Util.escapeMarkdown(content, true)}\n\`\`\``;
+        if (isSplit) {
+          splitOptions.prepend = `${splitOptions.prepend || ''}\`\`\`${codeName}\n`;
+          splitOptions.append = `\n\`\`\`${splitOptions.append || ''}`;
+        }
+      } else if (mentionPart) {
+        content = `${mentionPart}${content}`;
+      }
+
+      const disableEveryone = typeof this.options.disableEveryone === 'undefined' ?
+        this.target.client.options.disableEveryone :
+        this.options.disableEveryone;
+      if (disableEveryone) {
+        content = content.replace(/@(everyone|here)/g, '@\u200b$1');
+      }
+
+      if (isSplit) {
+        content = Util.splitMessage(content, splitOptions);
+      }
+    }
+
+    return content;
+  }
+
+  /**
+   * Resolves both data and files.
+   * @returns {APIMessage}
+   */
+  async resolve() {
+    this.resolveData();
+    await this.resolveFiles();
+    return this;
+  }
+
+  /**
+   * Resolves data.
+   * @returns {APIMessage}
+   */
+  resolveData() {
+    if (this.data) {
+      return this;
+    }
+
+    const content = this.makeContent();
+    const tts = Boolean(this.options.tts);
+    let nonce;
+    if (typeof this.options.nonce !== 'undefined') {
+      nonce = parseInt(this.options.nonce);
+      if (isNaN(nonce) || nonce < 0) throw new RangeError('MESSAGE_NONCE_TYPE');
+    }
+
+    const embedLikes = [];
+    if (this.isWebhook) {
```

Comment:
To avoid running this getter thrice (two here, one in resolveFiles), you could define this as a constant and re-use its output.